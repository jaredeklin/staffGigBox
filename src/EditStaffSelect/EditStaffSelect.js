import React, { Component } from 'react';

export class EditStaffSelect extends Component {
  constructor(props) {
    super(props);

    this.state = {
      staff_id: null,
      event_id: this.props.event_id
    }
  }

  handleChange = async (event) => {
    // console.log(parseInt(event.target.value))
    await this.setState({
      staff_id: event.target.value
    })

    this.props.updateEventStaff(this.state)
  }

  displayStaff = () => {
    this.props.staff.map(person => {
      // console.log('map', person)
      return (
        <option key={person.id}>
          { person.name }
        </option>
      )
    })
  }

  render() {

    const displayStaff = this.props.staff.map(person => {
      return (
        <option
          key={ person.id }
          value={ person.id }
        >
          { person.name }
        </option>
      )
    })
    return (
      <form>
        <select onChange={ this.handleChange }>
          { displayStaff }
        </select>
      </form>
    )
  }
}
