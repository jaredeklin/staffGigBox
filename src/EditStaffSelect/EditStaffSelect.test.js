import React from 'react';
import { shallow } from 'enzyme';
import { EditStaffSelect } from './EditStaffSelect';

describe('EditStaffSelect', () => {

  let wrapper;
  let mockStaff = [{ id: 2, name: 'Jared' }]
  let mockEventID = 1
  let mockUpdateEventStaff = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <EditStaffSelect 
        staff={ mockStaff }
        event_id={ mockEventID }
        updateEventStaff={ mockUpdateEventStaff }
      />
    )
  })

  it('should match the snapshot' , () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('handleChange should update state and call updateEventStaff', async() => {
    const mockEvent = { target: { value: 'oh yeah' }}

    await wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('staff_id')).toEqual('oh yeah')
    expect(mockUpdateEventStaff).toHaveBeenCalledWith(wrapper.state())
  })

})