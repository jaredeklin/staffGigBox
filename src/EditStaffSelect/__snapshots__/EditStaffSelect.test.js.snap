// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditStaffSelect should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditStaffSelect
    event_id={1}
    staff={
        Array [
            Object {
              "id": 2,
              "name": "Jared",
            },
          ]
    }
    updateEventStaff={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
>
        <option>
                Select staff
        </option>
        <option
                value={2}
        >
                Jared
        </option>
</select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Select staff
</option>,
          Array [
            <option
              value={2}
>
              Jared
</option>,
          ],
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select staff",
          },
          "ref": null,
          "rendered": "Select staff",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Jared",
            "value": 2,
          },
          "ref": null,
          "rendered": "Jared",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
>
          <option>
                    Select staff
          </option>
          <option
                    value={2}
          >
                    Jared
          </option>
</select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option>
              Select staff
</option>,
            Array [
              <option
                value={2}
>
                Jared
</option>,
            ],
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select staff",
            },
            "ref": null,
            "rendered": "Select staff",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Jared",
              "value": 2,
            },
            "ref": null,
            "rendered": "Jared",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
